delvingbitcoin_2_elasticsearch/delvingbitcoin_2_elasticsearch.py

```diff
@@ -0,0 +1,76 @@
+import os
+import json
+from dotenv import load_dotenv
+
+# from loguru import logger
+from elastic import create_index, document_add, document_exist, document_view
+from achieve import download_dumps
+
+dotenv_path = os.path.join(os.path.dirname(__file__),'..','.env')
+load_dotenv(dotenv_path)
+
+# Get environment variables for path and index name
+INDEX = os.getenv("INDEX")
+ARCHIVE = os.getenv("ARCHIVE") or "archive"
+SUB_ARCHIVE = os.getenv("SUB_ARCHIVE") or "posts"
+
+# Create Index if it doesn't exist
+if create_index(INDEX):
+    print(f"Index: {INDEX}, created successfully.")
+else:
+    print(f"Index: {INDEX}, already exist.")
+
+# Get the directory where the Python script is located
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+# Specify the path to the folder containing JSON files
+folder_path = os.path.join(script_dir, ARCHIVE, SUB_ARCHIVE)
+
+
+def index_documents(files_path):
+    # Iterate through files in the specified path
+    for root, dirs, files in os.walk(files_path):
+        for file in files:
+            if file.endswith('.json'):
+                file_path = os.path.join(root, file)
+                print(f'Fetching document from {file_path}')
+
+                # Load JSON data from file
+                with open(file_path, 'r', encoding='utf-8') as json_file:
+                    document = json.load(json_file)
+
+                # Select required fields
+                doc = {
+                    'id': f'{document["id"]}_{document["username"]}_{document["topic_slug"]}_{document["post_number"]}',
+                    'authors': document['username'],
```

Comment:
this should be an array

Reply:
Here is an example from docs of bitcointranscripts doc in Elasticsearch. Fields like authors, tags, contributors are stored in  array and the rest are in string format, _so I'll update the same in the next commit._
```{
          "created_at": "2011-09-15T06:01:09.417",
          "title": "How are transactions in discarded forks merged back into the blockchain?",
          "body": """It sounds like transactions in discarded blocks are propagated back to the network to be included in the next block to be generated, is that the case?  If so, is that triggered in the client when it discards a fork and accepts the longer chain?""",
          "type": "question",
          "url": "https://bitcoin.stackexchange.com/questions/1075",
          "tags": [
            "transactions",
            "blockchain"
          ],
          "accepted_answer_id": "1076",
          "indexed_at": "2023-08-27T02:34:33.661626",
          "domain": "https://bitcoin.stackexchange.com",
          "body_type": "raw",
          "id": "stackexchange-1075",
          "authors": [
            "Kurtosis"
          ]
        }
      }